Added functionality to post a review

JGon26 opened the pull request:
@Lenaciousd @matthewtboyle @Cora-Du 

PR consists of the added functionality of sending in a review. This is an mvp right now. I have not added restrictions yet on not being able to send review without all mandatory fields entered. To send a review currently, every single field has to be entered. Also have not added the auto refresh, so the review will only appear  on the next load of the product

JGon26 committed:
got some work done on review modal

JGon26 committed:
significant progress to submit modal

JGon26 committed:
I am able to post from modal now

JGon26 committed:
Merge branch 'master' into reviews

Lenaciousd reviewed (commented):
When I submitted a review, I saw it! Nice Work! A few things to consider. One is that there are product info and put requests being console.logged, please remove them or comment them. Also, When I clicked submit on my review, the form stayed up. You may want to have the form vanish onClick.

JGon26 committed:
removed a couple console.logs

JGon26 commented:
@Lenaciousd 

Great catch! just removed all unnecessary console logs. Just updated to get modal to close on successful review submission. There are more improvements to be made, but Feel like it is in a state now to be merged

JGon26 committed:
Got the modal to close on successful review submission

Lenaciousd reviewed (commented):
I am getting an error on submit ( xhr.js:187 POST http://localhost:3001/reviews 404 (Not Found), and (Uncaught (in promise) Error: Request failed with status code 404 at createError (createError.js:16) at settle (settle.js:17) at XMLHttpRequest.onloadend (xhr.js:54)). Also, this may be related to the error but on clicking submit the form is not disappearing.
